todo_app/data/trello_items.py

```diff
@@ -0,0 +1,160 @@
+import requests
+import os
+import certifi
+import ssl
+from todo_app.flask_config import Config
+
+class Item:
+    def __init__(self, id, name, status = 'To Do'):
+        self.id = id
+        self.name = name
+        self.status = status
+
+    @classmethod
+    def from_trello_card(cls, card, list):
+        return cls(card['id'], card['name'], list["name"])
+    
+    def  __str__(self):
+            return f"{self.name}  - status: {self.status}"
+
+
+
+def get_items(config: Config) -> list[Item]:
+    """
+    Fetches all cards from the trello board with id 'boardId and converts them to items.
+    
+    Args:
+        config: User specific configuration parameters
+    
+    Returns:
+        list: The list of saved items.
+    """
+    userKey= config["TRELLO_API_KEY"]
+    userToken= config["TRELLO_API_TOKEN"]
+    boardId= config["TRELLO_BOARD_ID"]
+    
+    get_cards_url = f"http://api.trello.com/1/boards/{boardId}/cards"
+    query = {
+    'key': userKey,
+    'token': userToken
+    }
+    get_cards_response = requests.request(
+    "GET",
+    get_cards_url,
+    params=query,
+    verify=False)
+    
+    get_lists_url = f"https://api.trello.com/1/cards/{boardId}/list"
+    get_lists_response = requests.request(
+    "GET",
+    get_lists_url,
+    params=query,
+    verify=False)
+    
+    if get_cards_response.status_code != 200:
+        raise ValueError(f"Server responded {get_cards_response.status_code} instead of 200 OK") 
+    #response.json() returns a list of dictionaries - one dictionary for each card
+    #we can just get the id and name and create an "item" that we can easily display
+
+    items = []
+    for card in get_cards_response.json():
+        cardId = card["id"]
+        get_lists_url = f"https://api.trello.com/1/cards/{cardId}/list"
+        get_lists_response = requests.request(
+        "GET",
+        get_lists_url,
+        params=query,
+        verify=False)
+        items.append(Item.from_trello_card(card, get_lists_response.json()))
+    return items
+
+
+def add_item(config:Config, title:str):
+    """
+    Fetches the saved item with the specified ID.
+
+    Args:
+        config: User specific configuration parameters
+        id: The ID of the item.
+
+    Returns:
+        item: The saved item, or None if no items match the specified ID.
+    """
+    userKey= config["TRELLO_API_KEY"]
+    userToken= config["TRELLO_API_TOKEN"]
+    boardId= config["TRELLO_BOARD_ID"]
+    
+    get_lists_url = f"https://api.trello.com/1/boards/{boardId}/lists"
+    get_lists_query = {
+    'key': userKey,
+    'token': userToken,
+    }
+    get_lists_response = requests.request(
+    "GET",
+    get_lists_url,
+    params=get_lists_query,
+    verify=False)
+    
+    list_id = None
+    for trello_list in get_lists_response.json():
+        if trello_list["name"] == "New Items":
+            list_id = trello_list["id"]
```

Comment:
This is a great solution for finding the list id, but worth comparing whether it's worth doing, compared to the lazy approach - just ask the dev to find it for you, and set it as an environment variable!

I think lines 87-101 could all be replaced by:
```
os.getenv("TRELLO_TODO_ID")
```
This wouldn't be as dynamic and would mean a small amount of extra setup, but is weighed up against a simpler codebase and a quicker response time for the function